classes/blocks/class-articles.php

```diff
@@ -162,110 +115,52 @@ public function prepare_data( $fields ): array {
 		// b. inside post - Get results excluding specific post.
 		// 3) specific posts - Get posts by ids specified in backend - new behavior / manual override.
 		// 4) issue page - Get posts based on page's tags.
-		$fiedls_diff = count( array_diff( [ 'post_types', 'posts' ], array_keys( $fields ) ) );
-		if ( is_tag() && ! empty( $fields['tags'] ) && 2 === $fiedls_diff ) {
-			$args = $this->filter_posts_for_tag_page( $fields );
+		$fields_diff = count( array_diff( [ 'post_types', 'posts' ], array_keys( $fields ) ) );
+		if ( is_tag() && ! empty( $fields['tags'] ) && 2 === $fields_diff ) {
+			$args = self::filter_posts_for_tag_page( $fields );
 		} elseif ( ! empty( $fields['post_types'] ) ||
 				! empty( $fields['tags'] ) ||
-				! empty( $exclude_post_id ) ) {
-			$args = $this->filter_posts_by_page_types_or_tags( $fields );
+				! empty( $fields['exclude_post_id'] ) ) {
+			$args = self::filter_posts_by_page_types_or_tags( $fields );
```

Comment:
Can you also change the functions to static? Calling non static functions with self:: works, but can cause warnings in logs or IDE.